test/__mocks__/pacote/index.js

```diff
@@ -0,0 +1,9 @@
+const pacote = jest.genMockFromModule('pacote');
```

Comment:
tests complain 

```
ENOENT: no such file or directory, open 'node:fs/promises'
```

can you have a look?

Reply:
It seems very weird to me. In my investigation, all the tests are running fine if I am not installing the `pacote` module and keeping all other things untouched.

When I run `npm i pacote` before running the tests, I get the same error. I need to figure out the possible reason.

Reply:
The node version is the culprit here: https://github.com/asyncapi/generator/pull/1115#discussion_r1529957193

Also, several dependencies are deprecated, which results in failing tests. The project needs to be upgraded to a newer node version with updated dependencies. I suggest updating all the project dependencies to updated versions based on our requirements.

WDYT @derberg How should we move forward?